vars/buildDockerImage.groovy

```diff
@@ -0,0 +1,87 @@
+def call(String imageName, Map config=[:]) {
+  if (!config.registry) {
+    if (infra.isTrusted()) {
+      config.registry = "jenkinsciinfra/"
+    } else {
+      config.registry = "jenkins4eval/"
+    }
+  }
+
+  pipeline {
+    agent {
+      label 'docker&&linux'
+    }
+
+    options {
+      disableConcurrentBuilds()
+      buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+      timeout(time: 60, unit: "MINUTES")
+      ansiColor("xterm")
+    }
+
+    stages {
+      stage("Build") {
+        steps {
+          script {
+            GIT_COMMIT_REV = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
```

Comment:
isn't this already added by declarative pipeline or is it too long?

Reply:
/shrug
is it? its not super googleable